src/appleseed.python/bindmaterial.cpp

```diff
@@ -47,10 +47,22 @@ using namespace std;
 namespace
 {
     auto_release_ptr<Material> create_material(
-        const string&   name,
-        const bpy::dict& params)
+        const std::string& material_type,
+        const string&      name,
+        const bpy::dict&   params)
     {
-        return GenericMaterialFactory().create(name.c_str(), bpy_dict_to_param_array(params));
+        MaterialFactoryRegistrar factories;
+        const IMaterialFactory* factory = factories.lookup(material_type.c_str());
+
+        if (factory)
+            return factory->create(name.c_str(), bpy_dict_to_param_array(params));
+        else
+        {
+            PyErr_SetString(PyExc_RuntimeError, "Material type not found");
```

Comment:
Rebased. Fixed comments and replaced 'type' by 'model' in the rest of python constructors.